src/Common/SipHash.h

```diff
@@ -125,13 +130,13 @@ class SipHash
         current_word = 0;
         switch (end - data)
         {
-            case 7: current_bytes[6] = data[6]; [[fallthrough]];
-            case 6: current_bytes[5] = data[5]; [[fallthrough]];
-            case 5: current_bytes[4] = data[4]; [[fallthrough]];
-            case 4: current_bytes[3] = data[3]; [[fallthrough]];
-            case 3: current_bytes[2] = data[2]; [[fallthrough]];
-            case 2: current_bytes[1] = data[1]; [[fallthrough]];
-            case 1: current_bytes[0] = data[0]; [[fallthrough]];
+            case 7: current_bytes[CURRENT_BYTES_IDX(6)] = data[6]; [[fallthrough]];
```

Comment:
Maybe it will be better with builtin bswap and inline_memcpy.

Reply:
Since the related code deals with different lengths depending the remained data, I think the current code is more straightfoward than using bswap_32 and bswap_64.